shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1032,12 +1146,25 @@ - (void)setEditableTransform:(NSArray*)matrix {
 // candidates view for multi-stage input methods (e.g., Japanese) when using a
 // physical keyboard.
 
+- (void)setSelectionRects:(NSArray*)rects {
+  NSMutableArray* rectsAsRect = [[NSMutableArray alloc] initWithCapacity:[rects count]];
+  for (NSUInteger i = 0; i < [rects count]; i++) {
+    NSArray* rect = rects[i];
+    [rectsAsRect addObject:@[
+      [NSNumber numberWithFloat:[rect[0] floatValue]],
+      [NSNumber numberWithFloat:[rect[1] floatValue]],
+      [NSNumber numberWithFloat:[rect[2] floatValue]],
+      [NSNumber numberWithFloat:[rect[3] floatValue]], [NSNumber numberWithInt:[rect[4] intValue]]
+    ]];
+  }
+  _selectionRects = rectsAsRect;
```

Comment:
This is leaking, making it a property as I mentioned above will help you fix that.

Reply:
You need to actually used this as a property to get the memory management benefits `self.selectionRects = rectsAsRect;`

Reply:
So, the issue is that this is the setter for that property – so if I reference it as `self.selectionRects`, I get an error about `_selectionRects` not being referenced:

```
../../flutter/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm:1182:1: error: ivar '_selectionRects' which backs the property is not referenced in this property's accessor [-Werror,-Wunused-property-ivar]```

So, it looks to me like `setSelectedTextRangeLocal` does a similar thing where it sets `_selectedTextRange` directly, so I'm not sure what the right thing to do is here – perhaps I just need to rename this method to not override the setter?

Reply:
Would adding `autorelease` to the `rectsAsRect` declaration help?